guides/bulk.md

```diff
@@ -0,0 +1,271 @@
+
+Alternatively, the `bulk` method can accept an array of hashes where each hash represents a single operation. The following code is equivalent to the previous example:
+
+```rust
+let mut ops = BulkOperations::new();
+ops.push(
+    BulkOperation::index(json!({
+        "title": "Beauty and the Beast",
+        "year": "1991"
+    }))
+    .index(movies)
+    .id("1"),
+)?;
+ops.push(
+    BulkOperation::index(json!({
+        "title": "Beauty and the Beast - Live Action",
+        "year": "2017"
+    }))
+    .index(movies)
+    .id("2"),
+)?;
+ops.push(
+    BulkOperation::index(json!({
+        "title": "The Lion King",
+        "year": "1994"
+    }))
+    .index(books)
+    .id("1"),
+)?;
+
+client.bulk(BulkParts::None).body(vec![ops]).send().await?;
+```
+
+We will use this format for the rest of the examples in this guide.
+
+### Creating multiple documents
+
+Similarly, instead of calling the `create` method for each document, you can use the `bulk` API to create multiple documents in a single request. The following code creates three documents in the `movies` index and one in the `books` index:
+
+```rust
+let mut ops = BulkOperations::new();
+ops.push(
+    BulkOperation::create(json!({
+        "title": "Beauty and the Beast 2",
+        "year": "2030"
+    }))
+    .index(movies),
+)?;
+ops.push(
+    BulkOperation::create(json!({
+        "title": "Beauty and the Beast 3",
+        "year": "2031"
+    }))
+    .index(movies),
+)?;
+ops.push(
+    BulkOperation::create(json!({
+        "title": "Beauty and the Beast 4",
+        "year": "2049"
+    }))
+    .index(movies),
+)?;
+ops.push(
+    BulkOperation::create(json!({
+        "title": "The Lion King 2",
+        "year": "1998"
+    }))
+    .index(books),
+)?;
+client.bulk(BulkParts::None).body(vec![ops]).send().await?;
+```
+
+Note that we specified only the `_index` for the last document in the request body. This is because the `bulk` method accepts an `index` parameter that specifies the default `_index` for all bulk operations in the request body. Moreover, we omit the `_id` for each document and let OpenSearch generate them for us in this example, just like we can with the `create` method.
+
+### Updating multiple documents
+
+```rust
+let mut ops = BulkOperations::new();
+ops.push(
+    BulkOperation::update(json!({
+        "year": "1992"
+    }))
+    .index(movies)
+    .id("1"),
+)?;
+ops.push(
+    BulkOperation::update(json!({
+        "year": "2018"
+    }))
+    .index(movies)
+    .id("2"),
+)?;
+
+client.bulk(BulkParts::None).body(vec![ops]).send().await?;
+```
+
+### Deleting multiple documents
+
+```rust
+let mut ops = BulkOperations::new();
+ops.push(
+    BulkOperation::delete()
+    .index(movies)
+    .id("1"),
+)?;
+ops.push(
+    BulkOperation::delete()
+    .index(movies)
+    .id("2"),
+)?;
+
+client.bulk(BulkParts::None).body(vec![ops]).send().await?;
```

Comment:
The id gets specified in the `delete` call, and index can be moved to `bulk` call
```suggestion
let mut ops = BulkOperations::new();
ops.push(BulkOperation::<()>::delete("1"))?;
ops.push(BulkOperation::<()>::delete("2"))?;

client.bulk(BulkParts::Index(movies)).body(vec![ops]).send().await?;
```